// @elastic/eui/optimize/lib/services/index.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  keys: true,
  accessibleClickKeys: true,
  cascadingMenuKeys: true,
  comboBoxKeys: true,
  htmlIdGenerator: true,
  useGeneratedHtmlId: true,
  CENTER_ALIGNMENT: true,
  LEFT_ALIGNMENT: true,
  RIGHT_ALIGNMENT: true,
  CurrentEuiBreakpointContext: true,
  CurrentEuiBreakpointProvider: true,
  useCurrentEuiBreakpoint: true,
  useIsWithinBreakpoints: true,
  useIsWithinMaxBreakpoint: true,
  useIsWithinMinBreakpoint: true,
  brighten: true,
  calculateContrast: true,
  calculateLuminance: true,
  colorPalette: true,
  darken: true,
  DEFAULT_VISUALIZATION_COLOR: true,
  desaturate: true,
  euiPaletteColorBlind: true,
  euiPaletteColorBlindBehindText: true,
  euiPaletteComplimentary: true,
  euiPaletteCool: true,
  euiPaletteForDarkBackground: true,
  euiPaletteForLightBackground: true,
  euiPaletteForStatus: true,
  euiPaletteForTemperature: true,
  euiPaletteGray: true,
  euiPaletteNegative: true,
  euiPalettePositive: true,
  euiPaletteWarm: true,
  getSteppedGradient: true,
  hexToHsv: true,
  hexToRgb: true,
  hsvToHex: true,
  hsvToRgb: true,
  isColorDark: true,
  isValidHex: true,
  lightness: true,
  makeDisabledContrastColor: true,
  makeHighContrastColor: true,
  rgbToHex: true,
  rgbToHsv: true,
  saturate: true,
  shade: true,
  shadeOrTint: true,
  tint: true,
  tintOrShade: true,
  transparentize: true,
  VISUALIZATION_COLORS: true,
  wcagContrastMin: true,
  useColorPickerState: true,
  useColorStopsState: true,
  copyToClipboard: true,
  dateFormatAliases: true,
  formatAuto: true,
  formatBoolean: true,
  formatDate: true,
  formatNumber: true,
  formatText: true,
  isEvenlyDivisibleBy: true,
  isWithinRange: true,
  Pager: true,
  calculatePopoverPosition: true,
  findPopoverPosition: true,
  Random: true,
  getSecureRelForTarget: true,
  Comparators: true,
  PropertySortType: true,
  SortableProperties: true,
  SortDirection: true,
  SortDirectionType: true,
  slugify: true,
  toInitials: true,
  toSentenceCase: true,
  throttle: true,
  getDurationAndPerformOnFrame: true,
  getTransitionTimings: true,
  getWaitDuration: true,
  performOnFrame: true,
  EuiWindowEvent: true
};
Object.defineProperty(exports, "CENTER_ALIGNMENT", {
  enumerable: true,
  get: function get() {
    return _alignment.CENTER_ALIGNMENT;
  }
});
Object.defineProperty(exports, "Comparators", {
  enumerable: true,
  get: function get() {
    return _sort.Comparators;
  }
});
Object.defineProperty(exports, "CurrentEuiBreakpointContext", {
  enumerable: true,
  get: function get() {
    return _breakpoint.CurrentEuiBreakpointContext;
  }
});
Object.defineProperty(exports, "CurrentEuiBreakpointProvider", {
  enumerable: true,
  get: function get() {
    return _breakpoint.CurrentEuiBreakpointProvider;
  }
});
Object.defineProperty(exports, "DEFAULT_VISUALIZATION_COLOR", {
  enumerable: true,
  get: function get() {
    return _color.DEFAULT_VISUALIZATION_COLOR;
  }
});
Object.defineProperty(exports, "EuiWindowEvent", {
  enumerable: true,
  get: function get() {
    return _window_event.EuiWindowEvent;
  }
});
Object.defineProperty(exports, "LEFT_ALIGNMENT", {
  enumerable: true,
  get: function get() {
    return _alignment.LEFT_ALIGNMENT;
  }
});
Object.defineProperty(exports, "Pager", {
  enumerable: true,
  get: function get() {
    return _paging.Pager;
  }
});
Object.defineProperty(exports, "PropertySortType", {
  enumerable: true,
  get: function get() {
    return _sort.PropertySortType;
  }
});
Object.defineProperty(exports, "RIGHT_ALIGNMENT", {
  enumerable: true,
  get: function get() {
    return _alignment.RIGHT_ALIGNMENT;
  }
});
Object.defineProperty(exports, "Random", {
  enumerable: true,
  get: function get() {
    return _random.Random;
  }
});
Object.defineProperty(exports, "SortDirection", {
  enumerable: true,
  get: function get() {
    return _sort.SortDirection;
  }
});
Object.defineProperty(exports, "SortDirectionType", {
  enumerable: true,
  get: function get() {
    return _sort.SortDirectionType;
  }
});
Object.defineProperty(exports, "SortableProperties", {
  enumerable: true,
  get: function get() {
    return _sort.SortableProperties;
  }
});
Object.defineProperty(exports, "VISUALIZATION_COLORS", {
  enumerable: true,
  get: function get() {
    return _color.VISUALIZATION_COLORS;
  }
});
Object.defineProperty(exports, "accessibleClickKeys", {
  enumerable: true,
  get: function get() {
    return _accessibility.accessibleClickKeys;
  }
});
Object.defineProperty(exports, "brighten", {
  enumerable: true,
  get: function get() {
    return _color.brighten;
  }
});
Object.defineProperty(exports, "calculateContrast", {
  enumerable: true,
  get: function get() {
    return _color.calculateContrast;
  }
});
Object.defineProperty(exports, "calculateLuminance", {
  enumerable: true,
  get: function get() {
    return _color.calculateLuminance;
  }
});
Object.defineProperty(exports, "calculatePopoverPosition", {
  enumerable: true,
  get: function get() {
    return _popover.calculatePopoverPosition;
  }
});
Object.defineProperty(exports, "cascadingMenuKeys", {
  enumerable: true,
  get: function get() {
    return _accessibility.cascadingMenuKeys;
  }
});
Object.defineProperty(exports, "colorPalette", {
  enumerable: true,
  get: function get() {
    return _color.colorPalette;
  }
});
Object.defineProperty(exports, "comboBoxKeys", {
  enumerable: true,
  get: function get() {
    return _accessibility.comboBoxKeys;
  }
});
Object.defineProperty(exports, "copyToClipboard", {
  enumerable: true,
  get: function get() {
    return _copy_to_clipboard.copyToClipboard;
  }
});
Object.defineProperty(exports, "darken", {
  enumerable: true,
  get: function get() {
    return _color.darken;
  }
});
Object.defineProperty(exports, "dateFormatAliases", {
  enumerable: true,
  get: function get() {
    return _format.dateFormatAliases;
  }
});
Object.defineProperty(exports, "desaturate", {
  enumerable: true,
  get: function get() {
    return _color.desaturate;
  }
});
Object.defineProperty(exports, "euiPaletteColorBlind", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteColorBlind;
  }
});
Object.defineProperty(exports, "euiPaletteColorBlindBehindText", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteColorBlindBehindText;
  }
});
Object.defineProperty(exports, "euiPaletteComplimentary", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteComplimentary;
  }
});
Object.defineProperty(exports, "euiPaletteCool", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteCool;
  }
});
Object.defineProperty(exports, "euiPaletteForDarkBackground", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteForDarkBackground;
  }
});
Object.defineProperty(exports, "euiPaletteForLightBackground", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteForLightBackground;
  }
});
Object.defineProperty(exports, "euiPaletteForStatus", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteForStatus;
  }
});
Object.defineProperty(exports, "euiPaletteForTemperature", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteForTemperature;
  }
});
Object.defineProperty(exports, "euiPaletteGray", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteGray;
  }
});
Object.defineProperty(exports, "euiPaletteNegative", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteNegative;
  }
});
Object.defineProperty(exports, "euiPalettePositive", {
  enumerable: true,
  get: function get() {
    return _color.euiPalettePositive;
  }
});
Object.defineProperty(exports, "euiPaletteWarm", {
  enumerable: true,
  get: function get() {
    return _color.euiPaletteWarm;
  }
});
Object.defineProperty(exports, "findPopoverPosition", {
  enumerable: true,
  get: function get() {
    return _popover.findPopoverPosition;
  }
});
Object.defineProperty(exports, "formatAuto", {
  enumerable: true,
  get: function get() {
    return _format.formatAuto;
  }
});
Object.defineProperty(exports, "formatBoolean", {
  enumerable: true,
  get: function get() {
    return _format.formatBoolean;
  }
});
Object.defineProperty(exports, "formatDate", {
  enumerable: true,
  get: function get() {
    return _format.formatDate;
  }
});
Object.defineProperty(exports, "formatNumber", {
  enumerable: true,
  get: function get() {
    return _format.formatNumber;
  }
});
Object.defineProperty(exports, "formatText", {
  enumerable: true,
  get: function get() {
    return _format.formatText;
  }
});
Object.defineProperty(exports, "getDurationAndPerformOnFrame", {
  enumerable: true,
  get: function get() {
    return _transition.getDurationAndPerformOnFrame;
  }
});
Object.defineProperty(exports, "getSecureRelForTarget", {
  enumerable: true,
  get: function get() {
    return _security.getSecureRelForTarget;
  }
});
Object.defineProperty(exports, "getSteppedGradient", {
  enumerable: true,
  get: function get() {
    return _color.getSteppedGradient;
  }
});
Object.defineProperty(exports, "getTransitionTimings", {
  enumerable: true,
  get: function get() {
    return _transition.getTransitionTimings;
  }
});
Object.defineProperty(exports, "getWaitDuration", {
  enumerable: true,
  get: function get() {
    return _transition.getWaitDuration;
  }
});
Object.defineProperty(exports, "hexToHsv", {
  enumerable: true,
  get: function get() {
    return _color.hexToHsv;
  }
});
Object.defineProperty(exports, "hexToRgb", {
  enumerable: true,
  get: function get() {
    return _color.hexToRgb;
  }
});
Object.defineProperty(exports, "hsvToHex", {
  enumerable: true,
  get: function get() {
    return _color.hsvToHex;
  }
});
Object.defineProperty(exports, "hsvToRgb", {
  enumerable: true,
  get: function get() {
    return _color.hsvToRgb;
  }
});
Object.defineProperty(exports, "htmlIdGenerator", {
  enumerable: true,
  get: function get() {
    return _accessibility.htmlIdGenerator;
  }
});
Object.defineProperty(exports, "isColorDark", {
  enumerable: true,
  get: function get() {
    return _color.isColorDark;
  }
});
Object.defineProperty(exports, "isEvenlyDivisibleBy", {
  enumerable: true,
  get: function get() {
    return _number.isEvenlyDivisibleBy;
  }
});
Object.defineProperty(exports, "isValidHex", {
  enumerable: true,
  get: function get() {
    return _color.isValidHex;
  }
});
Object.defineProperty(exports, "isWithinRange", {
  enumerable: true,
  get: function get() {
    return _number.isWithinRange;
  }
});
exports.keys = void 0;
Object.defineProperty(exports, "lightness", {
  enumerable: true,
  get: function get() {
    return _color.lightness;
  }
});
Object.defineProperty(exports, "makeDisabledContrastColor", {
  enumerable: true,
  get: function get() {
    return _color.makeDisabledContrastColor;
  }
});
Object.defineProperty(exports, "makeHighContrastColor", {
  enumerable: true,
  get: function get() {
    return _color.makeHighContrastColor;
  }
});
Object.defineProperty(exports, "performOnFrame", {
  enumerable: true,
  get: function get() {
    return _transition.performOnFrame;
  }
});
Object.defineProperty(exports, "rgbToHex", {
  enumerable: true,
  get: function get() {
    return _color.rgbToHex;
  }
});
Object.defineProperty(exports, "rgbToHsv", {
  enumerable: true,
  get: function get() {
    return _color.rgbToHsv;
  }
});
Object.defineProperty(exports, "saturate", {
  enumerable: true,
  get: function get() {
    return _color.saturate;
  }
});
Object.defineProperty(exports, "shade", {
  enumerable: true,
  get: function get() {
    return _color.shade;
  }
});
Object.defineProperty(exports, "shadeOrTint", {
  enumerable: true,
  get: function get() {
    return _color.shadeOrTint;
  }
});
Object.defineProperty(exports, "slugify", {
  enumerable: true,
  get: function get() {
    return _string.slugify;
  }
});
Object.defineProperty(exports, "throttle", {
  enumerable: true,
  get: function get() {
    return _throttle.throttle;
  }
});
Object.defineProperty(exports, "tint", {
  enumerable: true,
  get: function get() {
    return _color.tint;
  }
});
Object.defineProperty(exports, "tintOrShade", {
  enumerable: true,
  get: function get() {
    return _color.tintOrShade;
  }
});
Object.defineProperty(exports, "toInitials", {
  enumerable: true,
  get: function get() {
    return _string.toInitials;
  }
});
Object.defineProperty(exports, "toSentenceCase", {
  enumerable: true,
  get: function get() {
    return _string.toSentenceCase;
  }
});
Object.defineProperty(exports, "transparentize", {
  enumerable: true,
  get: function get() {
    return _color.transparentize;
  }
});
Object.defineProperty(exports, "useColorPickerState", {
  enumerable: true,
  get: function get() {
    return _color_picker.useColorPickerState;
  }
});
Object.defineProperty(exports, "useColorStopsState", {
  enumerable: true,
  get: function get() {
    return _color_picker.useColorStopsState;
  }
});
Object.defineProperty(exports, "useCurrentEuiBreakpoint", {
  enumerable: true,
  get: function get() {
    return _breakpoint.useCurrentEuiBreakpoint;
  }
});
Object.defineProperty(exports, "useGeneratedHtmlId", {
  enumerable: true,
  get: function get() {
    return _accessibility.useGeneratedHtmlId;
  }
});
Object.defineProperty(exports, "useIsWithinBreakpoints", {
  enumerable: true,
  get: function get() {
    return _breakpoint.useIsWithinBreakpoints;
  }
});
Object.defineProperty(exports, "useIsWithinMaxBreakpoint", {
  enumerable: true,
  get: function get() {
    return _breakpoint.useIsWithinMaxBreakpoint;
  }
});
Object.defineProperty(exports, "useIsWithinMinBreakpoint", {
  enumerable: true,
  get: function get() {
    return _breakpoint.useIsWithinMinBreakpoint;
  }
});
Object.defineProperty(exports, "wcagContrastMin", {
  enumerable: true,
  get: function get() {
    return _color.wcagContrastMin;
  }
});
var keys = _interopRequireWildcard(require("./keys"));
exports.keys = keys;
var _accessibility = require("./accessibility");
var _alignment = require("./alignment");
var _breakpoint = require("./breakpoint");
var _color = require("./color");
var _color_picker = require("./color_picker");
var _console = require("./console");
Object.keys(_console).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _console[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _console[key];
    }
  });
});
var _copy_to_clipboard = require("./copy_to_clipboard");
var _findElement = require("./findElement");
Object.keys(_findElement).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _findElement[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _findElement[key];
    }
  });
});
var _format = require("./format");
var _hooks = require("./hooks");
Object.keys(_hooks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hooks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _hooks[key];
    }
  });
});
var _number = require("./number");
var _paging = require("./paging");
var _popover = require("./popover");
var _random = require("./random");
var _security = require("./security");
var _sort = require("./sort");
var _string = require("./string");
var _theme = require("./theme");
Object.keys(_theme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _theme[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _theme[key];
    }
  });
});
var _throttle = require("./throttle");
var _transition = require("./transition");
var _window_event = require("./window_event");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
