// @elastic/eui/optimize/lib/components/inline_edit/inline_edit_title.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiInlineEditTitleStyles = void 0;
var _react = require("@emotion/react");
var _title = require("../title/title.styles");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

var euiInlineEditTitleStyles = function euiInlineEditTitleStyles(euiThemeContext) {
  var titleFontStyles = (0, _title.euiTitleStyles)(euiThemeContext);
  return {
    euiInlineEditTitle: /*#__PURE__*/(0, _react.css)(";label:euiInlineEditTitle;"),
    fontSize: {
      xxxs: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.xxxs, ";};label:xxxs;"),
      xxs: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.xxs, ";};label:xxs;"),
      xs: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.xs, ";};label:xs;"),
      s: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.s, ";};label:s;"),
      m: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.m, ";};label:m;"),
      l: /*#__PURE__*/(0, _react.css)(".euiFieldText{", titleFontStyles.l, ";};label:l;")
    }
  };
};
exports.euiInlineEditTitleStyles = euiInlineEditTitleStyles;
