// @elastic/eui/optimize/lib/components/inline_edit/inline_edit_text.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiInlineEditTextStyles = void 0;
var _react = require("@emotion/react");
var _text = require("../text/text.styles");
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

var euiInlineEditTextStyles = function euiInlineEditTextStyles(euiThemeContext) {
  var textFontStyles = (0, _text.euiTextStyles)(euiThemeContext);
  return {
    euiInlineEditText: /*#__PURE__*/(0, _react.css)(";label:euiInlineEditText;"),
    fontSize: {
      xs: /*#__PURE__*/(0, _react.css)(".euiFieldText{", textFontStyles.xs, ";};label:xs;"),
      s: /*#__PURE__*/(0, _react.css)(".euiFieldText{", textFontStyles.s, ";};label:s;"),
      m: /*#__PURE__*/(0, _react.css)(".euiFieldText{", textFontStyles.m, ";};label:m;")
    }
  };
};
exports.euiInlineEditTextStyles = euiInlineEditTextStyles;
