// @elastic/eui/lib/test/rtl/data_test_subj_queries.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.queryByTestSubject = exports.queryAllByTestSubject = exports.getByTestSubject = exports.getAllByTestSubject = exports.findByTestSubject = exports.findAllByTestSubject = void 0;
var _react = require("@testing-library/react");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"]; if (null != _i) { var _s, _e, _x, _r, _arr = [], _n = !0, _d = !1; try { if (_x = (_i = _i.call(arr)).next, 0 === i) { if (Object(_i) !== _i) return; _n = !1; } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0); } catch (err) { _d = !0, _e = err; } finally { try { if (!_n && null != _i.return && (_r = _i.return(), Object(_r) !== _r)) return; } finally { if (_d) throw _e; } } return _arr; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; } /*
                                                                       * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
                                                                       * or more contributor license agreements. Licensed under the Elastic License
                                                                       * 2.0 and the Server Side Public License, v 1; you may not use this file except
                                                                       * in compliance with, at your election, the Elastic License 2.0 or the Server
                                                                       * Side Public License, v 1.
                                                                       */
var queryAllByTestSubject = function queryAllByTestSubject() {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }
  return _react.queryHelpers.queryAllByAttribute.apply(_react.queryHelpers, ['data-test-subj'].concat(args));
};
exports.queryAllByTestSubject = queryAllByTestSubject;
var getMultipleError = function getMultipleError(_, TestSubjectValue) {
  return "Found multiple elements with the data-test-subj attribute of: ".concat(TestSubjectValue);
};
var getMissingError = function getMissingError(_, TestSubjectValue) {
  return "Unable to find an element with the data-test-subj attribute of: ".concat(TestSubjectValue);
};
var _buildQueries = (0, _react.buildQueries)(queryAllByTestSubject, getMultipleError, getMissingError),
  _buildQueries2 = _slicedToArray(_buildQueries, 5),
  queryByTestSubject = _buildQueries2[0],
  getAllByTestSubject = _buildQueries2[1],
  getByTestSubject = _buildQueries2[2],
  findAllByTestSubject = _buildQueries2[3],
  findByTestSubject = _buildQueries2[4];
exports.findByTestSubject = findByTestSubject;
exports.findAllByTestSubject = findAllByTestSubject;
exports.getByTestSubject = getByTestSubject;
exports.getAllByTestSubject = getAllByTestSubject;
exports.queryByTestSubject = queryByTestSubject;
