// @elastic/eui/test-env/components/flyout/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "EuiFlyout", {
  enumerable: true,
  get: function get() {
    return _flyout.EuiFlyout;
  }
});
Object.defineProperty(exports, "EuiFlyoutBody", {
  enumerable: true,
  get: function get() {
    return _flyout_body.EuiFlyoutBody;
  }
});
Object.defineProperty(exports, "EuiFlyoutFooter", {
  enumerable: true,
  get: function get() {
    return _flyout_footer.EuiFlyoutFooter;
  }
});
Object.defineProperty(exports, "EuiFlyoutHeader", {
  enumerable: true,
  get: function get() {
    return _flyout_header.EuiFlyoutHeader;
  }
});
Object.defineProperty(exports, "euiFlyoutSlideInLeft", {
  enumerable: true,
  get: function get() {
    return _flyout2.euiFlyoutSlideInLeft;
  }
});
Object.defineProperty(exports, "euiFlyoutSlideInRight", {
  enumerable: true,
  get: function get() {
    return _flyout2.euiFlyoutSlideInRight;
  }
});
var _flyout = require("./flyout");
var _flyout_body = require("./flyout_body");
var _flyout_footer = require("./flyout_footer");
var _flyout_header = require("./flyout_header");
var _flyout2 = require("./flyout.styles");
