// @elastic/eui/optimize/es/components/flyout/flyout_header.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS } from '../../global_styling';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "1hivny1-euiFlyoutHeader",
  styles: "flex-grow:0;label:euiFlyoutHeader;"
} : {
  name: "1hivny1-euiFlyoutHeader",
  styles: "flex-grow:0;label:euiFlyoutHeader;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiFlyoutHeaderStyles = function euiFlyoutHeaderStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return {
    euiFlyoutHeader: _ref,
    hasBorder: /*#__PURE__*/css(logicalCSS('border-bottom', euiTheme.border.thin), ";;label:hasBorder;")
  };
};
