// @elastic/eui/es/components/timeline/timeline_item.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "5bggde-euiTimelineItem",
  styles: "display:flex;position:relative;label:euiTimelineItem;"
} : {
  name: "5bggde-euiTimelineItem",
  styles: "display:flex;position:relative;label:euiTimelineItem;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiTimelineItemStyles = function euiTimelineItemStyles() {
  return {
    euiTimelineItem: _ref,
    // Vertical alignments
    // These classes are being targeted in timeline.styles.ts
    top: /*#__PURE__*/css(";label:top;"),
    center: /*#__PURE__*/css(";label:center;")
  };
};
