// @elastic/eui/es/components/timeline/timeline_item_event.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "16qbu87-center",
  styles: "align-self:center;label:center;"
} : {
  name: "16qbu87-center",
  styles: "align-self:center;label:center;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "1vysov7-top",
  styles: "align-self:flex-start;label:top;"
} : {
  name: "1vysov7-top",
  styles: "align-self:flex-start;label:top;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref3 = process.env.NODE_ENV === "production" ? {
  name: "1ozjlze-euiTimelineItemEvent",
  styles: "flex:1;label:euiTimelineItemEvent;"
} : {
  name: "1ozjlze-euiTimelineItemEvent",
  styles: "flex:1;label:euiTimelineItemEvent;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiTimelineItemEventStyles = function euiTimelineItemEventStyles() {
  return {
    euiTimelineItemEvent: _ref3,
    //  Vertical alignments
    top: _ref2,
    center: _ref
  };
};
