// @elastic/eui/optimize/lib/components/steps/steps_horizontal.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiStepsHorizontalStyles = void 0;
var _react = require("@emotion/react");
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; } /*
                                                                                                                                                                                                                                                                                           * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
                                                                                                                                                                                                                                                                                           * or more contributor license agreements. Licensed under the Elastic License
                                                                                                                                                                                                                                                                                           * 2.0 and the Server Side Public License, v 1; you may not use this file except
                                                                                                                                                                                                                                                                                           * in compliance with, at your election, the Elastic License 2.0 or the Server
                                                                                                                                                                                                                                                                                           * Side Public License, v 1.
                                                                                                                                                                                                                                                                                           */
var _ref = process.env.NODE_ENV === "production" ? {
  name: "1pmkwst-euiStepsHorizontal__item",
  styles: "flex-grow:1;flex-basis:0%;&:first-of-type>.euiStepHorizontal::before,&:last-of-type>.euiStepHorizontal::after{display:none;};label:euiStepsHorizontal__item;"
} : {
  name: "1pmkwst-euiStepsHorizontal__item",
  styles: "flex-grow:1;flex-basis:0%;&:first-of-type>.euiStepHorizontal::before,&:last-of-type>.euiStepHorizontal::after{display:none;};label:euiStepsHorizontal__item;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "44yl7r-euiStepsHorizontal",
  styles: "display:flex;align-items:stretch;label:euiStepsHorizontal;"
} : {
  name: "44yl7r-euiStepsHorizontal",
  styles: "display:flex;align-items:stretch;label:euiStepsHorizontal;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var euiStepsHorizontalStyles = function euiStepsHorizontalStyles() {
  return {
    euiStepsHorizontal: _ref2,
    euiStepsHorizontal__item: _ref
  };
};
exports.euiStepsHorizontalStyles = euiStepsHorizontalStyles;
