// @elastic/eui/optimize/es/components/page/page_section/page_section.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalCSS } from '../../../global_styling';
export var ALIGNMENTS = ['top', 'center', 'horizontalCenter'];
var _ref = process.env.NODE_ENV === "production" ? {
  name: "122lw0e-grow",
  styles: "flex-grow:1;label:grow;"
} : {
  name: "122lw0e-grow",
  styles: "flex-grow:1;label:grow;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiPageSectionStyles = function euiPageSectionStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    euiPageSection: /*#__PURE__*/css(logicalCSS('width', '100%'), logicalCSS('min-width', '0'), " display:flex;flex-direction:column;;label:euiPageSection;"),
    grow: _ref,
    border: /*#__PURE__*/css(logicalCSS('border-bottom', euiTheme.border.thin), ";;label:border;"),
    // Alignments
    top: /*#__PURE__*/css(";label:top;"),
    center: /*#__PURE__*/css(logicalCSS('width', 'auto'), " align-items:center;justify-content:center;;label:center;"),
    horizontalCenter: /*#__PURE__*/css(logicalCSS('width', 'auto'), " align-items:center;;label:horizontalCenter;")
  };
};
export var euiPageSectionContentStyles = function euiPageSectionContentStyles() {
  return {
    euiPageSection__content: /*#__PURE__*/css(logicalCSS('width', '100%'), ";;label:euiPageSection__content;"),
    // Alignments
    center: /*#__PURE__*/css(logicalCSS('width', 'auto'), " ", logicalCSS('margin-horizontal', 'auto'), ";;label:center;"),
    // Max widths
    restrictWidth: /*#__PURE__*/css(logicalCSS('margin-horizontal', 'auto'), ";;label:restrictWidth;")
  };
};
