// @elastic/eui/optimize/es/components/list_group/pinnable_list_group/pinnable_list_group.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { makeHighContrastColor } from '../../../services';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "1du20m7-euiPinnableListGroup__itemExtraAction",
  styles: "svg{transform:rotate(45deg);};label:euiPinnableListGroup__itemExtraAction;"
} : {
  name: "1du20m7-euiPinnableListGroup__itemExtraAction",
  styles: "svg{transform:rotate(45deg);};label:euiPinnableListGroup__itemExtraAction;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiPinnableListGroupItemExtraActionStyles = function euiPinnableListGroupItemExtraActionStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    // Base
    euiPinnableListGroup__itemExtraAction: _ref,
    // Variants
    pinned: /*#__PURE__*/css("&:not(:hover, :focus){color:", makeHighContrastColor(euiTheme.colors.lightShade)(euiTheme), ";};label:pinned;")
  };
};
