// @elastic/eui/es/components/form/range/range_thumb.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { mathWithUnits } from '../../../global_styling';
import { euiRangeThumbStyle, euiRangeThumbFocus, euiRangeVariables } from './range.styles';
import { euiCustomControl } from '../form.styles';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "9ugg7u-hasTicks",
  styles: "inset-block-start:0;margin-block-start:0;label:hasTicks;"
} : {
  name: "9ugg7u-hasTicks",
  styles: "inset-block-start:0;margin-block-start:0;label:hasTicks;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiRangeThumbStyles = function euiRangeThumbStyles(euiThemeContext) {
  var range = euiRangeVariables(euiThemeContext);
  return {
    // Base
    euiRangeThumb: /*#__PURE__*/css(euiCustomControl(euiThemeContext, {
      type: 'round'
    }), " ", euiRangeThumbStyle(euiThemeContext), " content:'';position:absolute;inset-inline-start:0;inset-block-start:50%;margin-block-start:", mathWithUnits(range.thumbHeight, function (x) {
      return x / 2 * -1;
    }), ";pointer-events:none;z-index:", range.thumbZIndex, ";&:focus{", euiRangeThumbFocus(euiThemeContext), " outline:none;};label:euiRangeThumb;"),
    hasTicks: _ref
  };
};
