// @elastic/eui/optimize/es/components/health/health.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { euiFontSize } from '../../global_styling';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "1qhojri-inherit",
  styles: "font-size:inherit;label:inherit;"
} : {
  name: "1qhojri-inherit",
  styles: "font-size:inherit;label:inherit;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "j8uhck-euiHealth",
  styles: "display:inline-block;label:euiHealth;"
} : {
  name: "j8uhck-euiHealth",
  styles: "display:inline-block;label:euiHealth;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiHealthStyles = function euiHealthStyles(euiTheme) {
  return {
    euiHealth: _ref2,
    // Text sizes
    xs: /*#__PURE__*/css(euiFontSize(euiTheme, 'xs'), ";;label:xs;"),
    s: /*#__PURE__*/css(euiFontSize(euiTheme, 's'), ";;label:s;"),
    m: /*#__PURE__*/css(euiFontSize(euiTheme, 'm'), ";;label:m;"),
    inherit: _ref
  };
};
