// @elastic/eui/optimize/es/components/skeleton/skeleton_rectangle.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { euiSkeletonGradientAnimation } from './utils';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "ubzie1-none",
  styles: "border-radius:0;label:none;"
} : {
  name: "ubzie1-none",
  styles: "border-radius:0;label:none;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiSkeletonRectangleStyles = function euiSkeletonRectangleStyles(euiThemeContext) {
  var euiTheme = euiThemeContext.euiTheme;
  return {
    euiSkeletonRectangle: /*#__PURE__*/css("display:block;", euiSkeletonGradientAnimation(euiThemeContext, {
      slideSize: '-75%',
      gradientSize: '350%'
    }), ";;label:euiSkeletonRectangle;"),
    // Border radius
    s: /*#__PURE__*/css("border-radius:", euiTheme.border.radius.small, ";;label:s;"),
    m: /*#__PURE__*/css("border-radius:", euiTheme.border.radius.medium, ";;label:m;"),
    none: _ref
  };
};
