// @elastic/eui/optimize/lib/components/header/header_breadcrumbs/header_breadcrumbs.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiHeaderBreadcrumbs = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _breadcrumbs = require("../../breadcrumbs");
var _header_breadcrumbs = require("./header_breadcrumbs.styles");
var _services = require("../../../services");
var _react2 = require("@emotion/react");
var _excluded = ["className", "breadcrumbs"];
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */
var EuiHeaderBreadcrumbs = function EuiHeaderBreadcrumbs(_ref) {
  var className = _ref.className,
    breadcrumbs = _ref.breadcrumbs,
    rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var classes = (0, _classnames.default)('euiHeaderBreadcrumbs', className);
  var euiTheme = (0, _services.useEuiTheme)();
  var styles = (0, _header_breadcrumbs.euiHeaderBreadcrumbsStyles)(euiTheme);
  var cssHeaderBreadcrumbStyles = [styles.euiHeaderBreadcrumbs];
  return (0, _react2.jsx)(_breadcrumbs.EuiBreadcrumbs, (0, _extends2.default)({
    max: 4,
    truncate: true,
    breadcrumbs: breadcrumbs,
    className: classes,
    css: cssHeaderBreadcrumbStyles,
    type: "application"
  }, rest));
};
exports.EuiHeaderBreadcrumbs = EuiHeaderBreadcrumbs;
