// @elastic/eui/lib/components/page/page_section/page_section.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiPageSectionStyles = exports.euiPageSectionContentStyles = exports.ALIGNMENTS = void 0;
var _react = require("@emotion/react");
var _global_styling = require("../../../global_styling");
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; } /*
                                                                                                                                                                                                                                                                                           * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
                                                                                                                                                                                                                                                                                           * or more contributor license agreements. Licensed under the Elastic License
                                                                                                                                                                                                                                                                                           * 2.0 and the Server Side Public License, v 1; you may not use this file except
                                                                                                                                                                                                                                                                                           * in compliance with, at your election, the Elastic License 2.0 or the Server
                                                                                                                                                                                                                                                                                           * Side Public License, v 1.
                                                                                                                                                                                                                                                                                           */
var ALIGNMENTS = ['top', 'center', 'horizontalCenter'];
exports.ALIGNMENTS = ALIGNMENTS;
var _ref = process.env.NODE_ENV === "production" ? {
  name: "122lw0e-grow",
  styles: "flex-grow:1;label:grow;"
} : {
  name: "122lw0e-grow",
  styles: "flex-grow:1;label:grow;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var euiPageSectionStyles = function euiPageSectionStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    euiPageSection: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('width', '100%'), (0, _global_styling.logicalCSS)('min-width', '0'), " display:flex;flex-direction:column;;label:euiPageSection;"),
    grow: _ref,
    border: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('border-bottom', euiTheme.border.thin), ";;label:border;"),
    // Alignments
    top: /*#__PURE__*/(0, _react.css)(";label:top;"),
    center: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('width', 'auto'), " align-items:center;justify-content:center;;label:center;"),
    horizontalCenter: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('width', 'auto'), " align-items:center;;label:horizontalCenter;")
  };
};
exports.euiPageSectionStyles = euiPageSectionStyles;
var euiPageSectionContentStyles = function euiPageSectionContentStyles() {
  return {
    euiPageSection__content: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('width', '100%'), ";;label:euiPageSection__content;"),
    // Alignments
    center: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('width', 'auto'), " ", (0, _global_styling.logicalCSS)('margin-horizontal', 'auto'), ";;label:center;"),
    // Max widths
    restrictWidth: /*#__PURE__*/(0, _react.css)((0, _global_styling.logicalCSS)('margin-horizontal', 'auto'), ";;label:restrictWidth;")
  };
};
exports.euiPageSectionContentStyles = euiPageSectionContentStyles;
