// @elastic/eui/lib/components/form/range/range_thumb.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiRangeThumbStyles = void 0;
var _react = require("@emotion/react");
var _global_styling = require("../../../global_styling");
var _range = require("./range.styles");
var _form = require("../form.styles");
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; } /*
                                                                                                                                                                                                                                                                                           * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
                                                                                                                                                                                                                                                                                           * or more contributor license agreements. Licensed under the Elastic License
                                                                                                                                                                                                                                                                                           * 2.0 and the Server Side Public License, v 1; you may not use this file except
                                                                                                                                                                                                                                                                                           * in compliance with, at your election, the Elastic License 2.0 or the Server
                                                                                                                                                                                                                                                                                           * Side Public License, v 1.
                                                                                                                                                                                                                                                                                           */
var _ref = process.env.NODE_ENV === "production" ? {
  name: "9ugg7u-hasTicks",
  styles: "inset-block-start:0;margin-block-start:0;label:hasTicks;"
} : {
  name: "9ugg7u-hasTicks",
  styles: "inset-block-start:0;margin-block-start:0;label:hasTicks;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var euiRangeThumbStyles = function euiRangeThumbStyles(euiThemeContext) {
  var range = (0, _range.euiRangeVariables)(euiThemeContext);
  return {
    // Base
    euiRangeThumb: /*#__PURE__*/(0, _react.css)((0, _form.euiCustomControl)(euiThemeContext, {
      type: 'round'
    }), " ", (0, _range.euiRangeThumbStyle)(euiThemeContext), " content:'';position:absolute;inset-inline-start:0;inset-block-start:50%;margin-block-start:", (0, _global_styling.mathWithUnits)(range.thumbHeight, function (x) {
      return x / 2 * -1;
    }), ";pointer-events:none;z-index:", range.thumbZIndex, ";&:focus{", (0, _range.euiRangeThumbFocus)(euiThemeContext), " outline:none;};label:euiRangeThumb;"),
    hasTicks: _ref
  };
};
exports.euiRangeThumbStyles = euiRangeThumbStyles;
