// @elastic/eui/lib/services/keys.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.keys = exports.TAB = exports.SPACE = exports.SHIFT = exports.PAGE_UP = exports.PAGE_DOWN = exports.META = exports.HOME = exports.F2 = exports.ESCAPE = exports.ENTER = exports.END = exports.CTRL = exports.BACKSPACE = exports.ARROW_UP = exports.ARROW_RIGHT = exports.ARROW_LEFT = exports.ARROW_DOWN = exports.ALT = void 0;
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

var ENTER = 'Enter';
exports.ENTER = ENTER;
var SPACE = ' ';
exports.SPACE = SPACE;
var ESCAPE = 'Escape';
exports.ESCAPE = ESCAPE;
var TAB = 'Tab';
exports.TAB = TAB;
var BACKSPACE = 'Backspace';
exports.BACKSPACE = BACKSPACE;
var F2 = 'F2';
exports.F2 = F2;
var ALT = 'Alt';
exports.ALT = ALT;
var SHIFT = 'Shift';
exports.SHIFT = SHIFT;
var CTRL = 'Control';
exports.CTRL = CTRL;
var META = 'Meta'; // Windows, Command, Option
exports.META = META;
var ARROW_DOWN = 'ArrowDown';
exports.ARROW_DOWN = ARROW_DOWN;
var ARROW_UP = 'ArrowUp';
exports.ARROW_UP = ARROW_UP;
var ARROW_LEFT = 'ArrowLeft';
exports.ARROW_LEFT = ARROW_LEFT;
var ARROW_RIGHT = 'ArrowRight';
exports.ARROW_RIGHT = ARROW_RIGHT;
var PAGE_UP = 'PageUp';
exports.PAGE_UP = PAGE_UP;
var PAGE_DOWN = 'PageDown';
exports.PAGE_DOWN = PAGE_DOWN;
var END = 'End';
exports.END = END;
var HOME = 'Home';
exports.HOME = HOME;
var keys = /*#__PURE__*/function (keys) {
  keys["ENTER"] = "Enter";
  keys["SPACE"] = " ";
  keys["ESCAPE"] = "Escape";
  keys["TAB"] = "Tab";
  keys["BACKSPACE"] = "Backspace";
  keys["F2"] = "F2";
  keys["ALT"] = "Alt";
  keys["SHIFT"] = "Shift";
  keys["CTRL"] = "Control";
  keys["META"] = "Meta";
  keys["ARROW_DOWN"] = "ArrowDown";
  keys["ARROW_UP"] = "ArrowUp";
  keys["ARROW_LEFT"] = "ArrowLeft";
  keys["ARROW_RIGHT"] = "ArrowRight";
  keys["PAGE_UP"] = "PageUp";
  keys["PAGE_DOWN"] = "PageDown";
  keys["END"] = "End";
  keys["HOME"] = "Home";
  return keys;
}({});
exports.keys = keys;
