// @elastic/eui/optimize/es/components/description_list/description_list.styles.js
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicalTextAlignCSS, euiMinBreakpoint } from '../../global_styling';
export var euiDescriptionListStyles = function euiDescriptionListStyles(euiThemeContext) {
  // Flex display for column and responsive column
  var columnDisplay = "\n    display: flex;\n    align-items: baseline;\n    flex-wrap: wrap;  \n  ";
  return {
    euiDescriptionList: /*#__PURE__*/css(";label:euiDescriptionList;"),
    // Types
    row: /*#__PURE__*/css(";label:row;"),
    inline: /*#__PURE__*/css(";label:inline;"),
    column: /*#__PURE__*/css(columnDisplay, ";;label:column;"),
    // Responsive columns behave as a row on breakpoints xs-s
    responsiveColumn: /*#__PURE__*/css(euiMinBreakpoint(euiThemeContext, 'm'), "{", columnDisplay, ";};label:responsiveColumn;"),
    // Alignment
    center: /*#__PURE__*/css(logicalTextAlignCSS('center'), ";;label:center;"),
    left: /*#__PURE__*/css(logicalTextAlignCSS('left'), ";;label:left;")
  };
};
