// @elastic/eui/optimize/es/components/list_group/list_group_item_extra_action.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { euiCanAnimate, logicalCSS } from '../../global_styling';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "ls7zqx-alwaysShow",
  styles: "opacity:1;label:alwaysShow;"
} : {
  name: "ls7zqx-alwaysShow",
  styles: "opacity:1;label:alwaysShow;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "3r3xc1-hoverStyles",
  styles: ".euiListGroupItem:hover &,.euiListGroupItem__button:focus+&,&:focus{opacity:1;};label:hoverStyles;"
} : {
  name: "3r3xc1-hoverStyles",
  styles: ".euiListGroupItem:hover &,.euiListGroupItem__button:focus+&,&:focus{opacity:1;};label:hoverStyles;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiListGroupItemExtraActionStyles = function euiListGroupItemExtraActionStyles(_ref3) {
  var euiTheme = _ref3.euiTheme;
  return {
    euiListGroupItemExtraAction: /*#__PURE__*/css("flex-shrink:0;opacity:0;", logicalCSS('margin-right', euiTheme.size.s), ";", euiCanAnimate, "{transition:opacity ", euiTheme.animation.fast, ";};label:euiListGroupItemExtraAction;"),
    hoverStyles: _ref2,
    alwaysShow: _ref
  };
};
