// @elastic/eui/optimize/lib/components/page/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "EuiPage", {
  enumerable: true,
  get: function get() {
    return _page.EuiPage;
  }
});
Object.defineProperty(exports, "EuiPageBody", {
  enumerable: true,
  get: function get() {
    return _page_body.EuiPageBody;
  }
});
Object.defineProperty(exports, "EuiPageContentBody_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentBody_Deprecated;
  }
});
Object.defineProperty(exports, "EuiPageContentHeaderSection_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentHeaderSection_Deprecated;
  }
});
Object.defineProperty(exports, "EuiPageContentHeader_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentHeader_Deprecated;
  }
});
Object.defineProperty(exports, "EuiPageContent_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContent_Deprecated;
  }
});
Object.defineProperty(exports, "EuiPageHeader", {
  enumerable: true,
  get: function get() {
    return _page_header.EuiPageHeader;
  }
});
Object.defineProperty(exports, "EuiPageHeaderContent", {
  enumerable: true,
  get: function get() {
    return _page_header.EuiPageHeaderContent;
  }
});
Object.defineProperty(exports, "EuiPageHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_header.EuiPageHeaderSection;
  }
});
Object.defineProperty(exports, "EuiPageSection", {
  enumerable: true,
  get: function get() {
    return _page_section.EuiPageSection;
  }
});
Object.defineProperty(exports, "EuiPageSideBar_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_side_bar.EuiPageSideBar_Deprecated;
  }
});
Object.defineProperty(exports, "EuiPageSidebar", {
  enumerable: true,
  get: function get() {
    return _page_sidebar.EuiPageSidebar;
  }
});
Object.defineProperty(exports, "EuiPageTemplate_Deprecated", {
  enumerable: true,
  get: function get() {
    return _page_template.EuiPageTemplate_Deprecated;
  }
});
var _page = require("./page");
var _page_body = require("./page_body");
var _page_content = require("./page_content");
var _page_header = require("./page_header");
var _page_section = require("./page_section");
var _page_side_bar = require("./page_side_bar");
var _page_sidebar = require("./page_sidebar");
var _page_template = require("./page_template");
