// @elastic/eui/test-env/test/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  requiredProps: true,
  takeMountedSnapshot: true,
  findTestSubject: true,
  startThrowingReactWarnings: true,
  stopThrowingReactWarnings: true,
  sleep: true
};
Object.defineProperty(exports, "findTestSubject", {
  enumerable: true,
  get: function get() {
    return _find_test_subject.findTestSubject;
  }
});
Object.defineProperty(exports, "requiredProps", {
  enumerable: true,
  get: function get() {
    return _required_props.requiredProps;
  }
});
Object.defineProperty(exports, "sleep", {
  enumerable: true,
  get: function get() {
    return _sleep.sleep;
  }
});
Object.defineProperty(exports, "startThrowingReactWarnings", {
  enumerable: true,
  get: function get() {
    return _react_warnings.startThrowingReactWarnings;
  }
});
Object.defineProperty(exports, "stopThrowingReactWarnings", {
  enumerable: true,
  get: function get() {
    return _react_warnings.stopThrowingReactWarnings;
  }
});
Object.defineProperty(exports, "takeMountedSnapshot", {
  enumerable: true,
  get: function get() {
    return _take_mounted_snapshot.takeMountedSnapshot;
  }
});
var _required_props = require("./required_props");
var _take_mounted_snapshot = require("./take_mounted_snapshot");
var _find_test_subject = require("./find_test_subject");
var _react_warnings = require("./react_warnings");
var _sleep = require("./sleep");
var _emotionPrefix = require("./emotion-prefix");
Object.keys(_emotionPrefix).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _emotionPrefix[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _emotionPrefix[key];
    }
  });
});
