// @elastic/eui/es/components/spacer/spacer.styles.js
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; }
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { logicals } from '../../global_styling';
var _ref = process.env.NODE_ENV === "production" ? {
  name: "191xi1n-euiSpacer",
  styles: "flex-shrink:0;label:euiSpacer;"
} : {
  name: "191xi1n-euiSpacer",
  styles: "flex-shrink:0;label:euiSpacer;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
export var euiSpacerStyles = function euiSpacerStyles(_ref2) {
  var euiTheme = _ref2.euiTheme;
  return {
    euiSpacer: _ref,
    // Sizes
    xs: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.xs, ";;label:xs;"),
    s: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.s, ";;label:s;"),
    m: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.base, ";;label:m;"),
    l: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.l, ";;label:l;"),
    xl: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.xl, ";;label:xl;"),
    xxl: /*#__PURE__*/css(logicals.height, ":", euiTheme.size.xxl, ";;label:xxl;")
  };
};
