// @elastic/eui/lib/components/list_group/list_group_item_extra_action.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.euiListGroupItemExtraActionStyles = void 0;
var _react = require("@emotion/react");
var _global_styling = require("../../global_styling");
function _EMOTION_STRINGIFIED_CSS_ERROR__() { return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop)."; } /*
                                                                                                                                                                                                                                                                                           * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
                                                                                                                                                                                                                                                                                           * or more contributor license agreements. Licensed under the Elastic License
                                                                                                                                                                                                                                                                                           * 2.0 and the Server Side Public License, v 1; you may not use this file except
                                                                                                                                                                                                                                                                                           * in compliance with, at your election, the Elastic License 2.0 or the Server
                                                                                                                                                                                                                                                                                           * Side Public License, v 1.
                                                                                                                                                                                                                                                                                           */
var _ref = process.env.NODE_ENV === "production" ? {
  name: "ls7zqx-alwaysShow",
  styles: "opacity:1;label:alwaysShow;"
} : {
  name: "ls7zqx-alwaysShow",
  styles: "opacity:1;label:alwaysShow;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "3r3xc1-hoverStyles",
  styles: ".euiListGroupItem:hover &,.euiListGroupItem__button:focus+&,&:focus{opacity:1;};label:hoverStyles;"
} : {
  name: "3r3xc1-hoverStyles",
  styles: ".euiListGroupItem:hover &,.euiListGroupItem__button:focus+&,&:focus{opacity:1;};label:hoverStyles;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
var euiListGroupItemExtraActionStyles = function euiListGroupItemExtraActionStyles(_ref3) {
  var euiTheme = _ref3.euiTheme;
  return {
    euiListGroupItemExtraAction: /*#__PURE__*/(0, _react.css)("flex-shrink:0;opacity:0;", (0, _global_styling.logicalCSS)('margin-right', euiTheme.size.s), ";", _global_styling.euiCanAnimate, "{transition:opacity ", euiTheme.animation.fast, ";};label:euiListGroupItemExtraAction;"),
    hoverStyles: _ref2,
    alwaysShow: _ref
  };
};
exports.euiListGroupItemExtraActionStyles = euiListGroupItemExtraActionStyles;
